apis/v1alpha2/shared_types.go

```diff
@@ -249,4 +238,41 @@ type RouteStatus struct {
 //
 // +kubebuilder:validation:MinLength=1
 // +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^(\*.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
```

Comment:
This matches IPv4 but not IPv6. Spec declares IPs are not allowd at all though..

Reply:
Is there a regex that could allow all valid hostnames but still invalidate IPv4 addresses? For reference, this is pulled from the upstream kubernetes wildcard subdomain validation (with the exception that the leading `*.` is optional here.

Reply:
I don't think there is, because IPv4 addresses are still technically valid domain names as well, sadly.

Reply:
I think trying to encode really complicated things into a regex will be problematic. It becomes quite hard to verify that a given regex does exactly what you want by looking at it. 

I would comment that this validation may not remove all cases and controller will need to check.